ophyd/ophydobj.py

```diff
@@ -104,10 +107,25 @@ def __init_subclass__(cls, version=None, version_of=None,
         super().__init_subclass__(**kwargs)
 
         if version is None:
+            if version_of is not None:
+                raise RuntimeError('Must specify a version if `version_of` '
+                                   'is specified')
+            if version_type is None:
+                return
+            # Allow specification of version_type without specifying a version,
+            # for use in a base class
+
+            cls._class_info_ = dict(
+                versions={},
+                version=None,
+                version_type=version_type,
+                version_of=version_of
+            )
             return
 
         if version_of is None:
             versions = {}
+            version_of = cls
         else:
             versions = version_of._class_info_['versions']
             version_type = version_of._class_info_['version_type']
```

Comment:
We should probably check that `version_type` was not also passed in (or if it was it matches it's parent)?

Reply:
👍